src/utils/Log.ts

```diff
@@ -0,0 +1,112 @@
+/**
```

Comment:
Sexy.

Reply:
I used a similar design in another project of mine and it ended up working really well and really cleaning up logging. Thought it'd be useful here too 😄